aclk/aclk_lws_wss_client.c

```diff
@@ -5,378 +5,376 @@
 }
 
-static const struct lws_protocols protocols[] = {
-	{
-		"aclk-wss",
-		aclk_lws_wss_callback,
-		sizeof(struct aclk_lws_wss_perconnect_data),
-		0, 0, 0, 0
-	},
-	{ NULL, NULL, 0, 0, 0, 0, 0 }
-};
+static const struct lws_protocols protocols[] = { { "aclk-wss", aclk_lws_wss_callback,
+                                                    sizeof(struct aclk_lws_wss_perconnect_data), 0, 0, 0, 0 },
+                                                  { NULL, NULL, 0, 0, 0, 0, 0 } };
 
-static void aclk_lws_wss_log_divert(int level, const char *line) {
-	switch(level){
-		case LLL_ERR:
-			error("Libwebsockets Error: %s", line);
-			break;
-		case LLL_WARN:
-			debug(D_ACLK, "Libwebsockets Warn: %s", line);
-			break;
-		default:
-			error("Libwebsockets try to log with unknown log level (%d), msg: %s", level, line);
-	}
+static void aclk_lws_wss_log_divert(int level, const char *line)
+{
+    switch (level) {
+        case LLL_ERR:
+            error("Libwebsockets Error: %s", line);
+            break;
+        case LLL_WARN:
+            debug(D_ACLK, "Libwebsockets Warn: %s", line);
+            break;
+        default:
+            error("Libwebsockets try to log with unknown log level (%d), msg: %s", level, line);
+    }
 }
 
-struct aclk_lws_wss_engine_instance* aclk_lws_wss_client_init (const struct aclk_lws_wss_engine_callbacks *callbacks, const char *target_hostname, int target_port) {
-	static int lws_logging_initialized = 0;
-	struct lws_context_creation_info info;
-	struct aclk_lws_wss_engine_instance *inst;
+static int aclk_lws_wss_client_init( const char *target_hostname, int target_port)
+{
+    static int lws_logging_initialized = 0;
+    struct lws_context_creation_info info;
 
-	if(unlikely(!lws_logging_initialized)) {
-		lws_set_log_level(LLL_ERR | LLL_WARN, aclk_lws_wss_log_divert);
-		lws_logging_initialized = 1;
-	}
+    if (unlikely(!lws_logging_initialized)) {
+        lws_set_log_level(LLL_ERR | LLL_WARN, aclk_lws_wss_log_divert);
+        lws_logging_initialized = 1;
+    }
 
-	if(!callbacks || !target_hostname)
-		return NULL;
+    if (!target_hostname)
+        return 1;
 
-	inst = callocz(1, sizeof(struct aclk_lws_wss_engine_instance));
+    engine_instance = callocz(1, sizeof(struct aclk_lws_wss_engine_instance));
 
-	inst->host = target_hostname;
-	inst->port = target_port;
+    engine_instance->host = target_hostname;
+    engine_instance->port = target_port;
 
-	memset(&info, 0, sizeof(struct lws_context_creation_info));
-	info.options = LWS_SERVER_OPTION_DO_SSL_GLOBAL_INIT;
-	info.port = CONTEXT_PORT_NO_LISTEN;
-	info.protocols = protocols;
-	info.user = inst;
-	
-	inst->lws_context = lws_create_context(&info);
-	if(!inst->lws_context)
-		goto failure_cleanup_2;
+    memset(&info, 0, sizeof(struct lws_context_creation_info));
+    info.options = LWS_SERVER_OPTION_DO_SSL_GLOBAL_INIT;
+    info.port = CONTEXT_PORT_NO_LISTEN;
+    info.protocols = protocols;
+    info.user = engine_instance;
```

Comment:
is this necessary if we remove
`struct aclk_lws_wss_engine_instance *inst = lws_context_user(lws_get_context(wsi));`

Reply:
That's true.